### ISSUE TITLE
 Enhance Social Interaction: Implement Friend and Block Features

### ISSUE THREAD

**Issue by GowthertG** [zwzone/ft_transcendence#80]
### Problem Statement:
The current state of the ping pong game lacks fundamental social features, notably the absence of a friend list and blocked list. The aim of this issue is to bridge this gap by incorporating essential backend functionality.

### Expected Behavior:
- Users should be able to seamlessly add friends and maintain a comprehensive list of their connections.
- The system must support the removal of friends from the list, providing users with control over their social network.
- Implement the ability to block and unblock users to manage interactions effectively.
- Users should have visibility into and management capabilities for their blocked list.

**Comment by zwzone** [comment]
I agree with friend list, but we will skip the feature of the block and unblock users
